mods/mc_core/coordinates.lua

```diff
@@ -0,0 +1,96 @@
+function mc_core.record_coordinates(player, message)
+	if mc_core.checkPrivs(player, {interact = true}) then
+		local pmeta = player:get_meta()
+		local pos = player:get_pos()
+		local realmID = pmeta:get_int("realm")
+		local temp, clean_message
+		clean_message = minetest.formspec_escape(message)
+		temp = minetest.deserialize(pmeta:get_string("coordinates")) or {realms = {}, coords = {}, note_map = {}}
+
+		if temp.realms and temp.coords then
+            table.insert(temp.realms, realmID)
+            table.insert(temp.coords, pos)
+        else
+            temp.realms = {realmID}
+            temp.coords = {pos}
+        end
+		temp.note_map[clean_message] = #temp.realms
+
+		pmeta:set_string("coordinates", minetest.serialize(temp))
+		minetest.chat_send_player(player:get_player_name(), minetest.colorize(mc_core.col.log, "[Minetest Classroom] Your position was recorded in your notebook."))
+	end
+end
+
+function mc_core.queue_marker(player, message, pos, expiry)
```

Comment:
Need a check in this function to ensure player has shout priv otherwise this could be used  to bypass a teacher mute.

Reply:
Good catch! Will add